Sources/OpenAPIRuntime/Base/CopyOnWriteBox.swift

```diff
@@ -0,0 +1,190 @@
+
+/// A reference type that wraps a value and enforces copy-on-write semantics.
+///
+/// It also enables recursive types by introducing a "box" into the cycle, which
+/// allows the owning type to have a finite size.
+@_spi(Generated)
+public final class CopyOnWriteBox<Wrapped> {
+
+    /// The stored value.
+    private var value: Wrapped
+
+    /// Creates a new box.
+    /// - Parameter value: The value to store in the box.
+    public init(value: Wrapped) {
+        self.value = value
+    }
+
+    /// Returns a copy of the stored value.
+    public func read() -> Wrapped {
+        value
+    }
+
+    /// Provides read-write access to the stored value and enforces
+    /// copy-on-write semantics.
+    /// - Parameters:
+    ///   - box: A reference to the existing box, into which a new reference
+    ///     might get written if a copy had to be made.
+    ///   - modify: The closure that modifies the value.
+    public static func write(
+        to box: inout CopyOnWriteBox<Wrapped>,
+        using modify: (inout Wrapped) -> Void
+    ) {
+        let resolvedBox: CopyOnWriteBox<Wrapped>
+        if isKnownUniquelyReferenced(&box) {
+            resolvedBox = box
+        } else {
+            resolvedBox = Self(value: box.value)
+            box = resolvedBox
+        }
+        modify(&resolvedBox.value)
+    }
+}
+
+extension CopyOnWriteBox: Encodable where Wrapped: Encodable {
+
+    /// Encodes this value into the given encoder.
+    ///
+    /// If the value fails to encode anything, `encoder` will encode an empty
+    /// keyed container in its place.
+    ///
+    /// This function throws an error if any values are invalid for the given
+    /// encoder's format.
+    ///
+    /// - Parameter encoder: The encoder to write data to.
+    /// - Throws: On an encoding error.
+    public func encode(to encoder: any Encoder) throws {
+        try value.encode(to: encoder)
+    }
+}
+
+extension CopyOnWriteBox: Decodable where Wrapped: Decodable {
+
+    /// Creates a new instance by decoding from the given decoder.
+    ///
+    /// This initializer throws an error if reading from the decoder fails, or
+    /// if the data read is corrupted or otherwise invalid.
+    ///
+    /// - Parameter decoder: The decoder to read data from.
+    /// - Throws: On a decoding error.
+    public convenience init(from decoder: any Decoder) throws {
+        let value = try Wrapped(from: decoder)
+        self.init(value: value)
+    }
+}
+
+extension CopyOnWriteBox: Equatable where Wrapped: Equatable {
+
+    /// Returns a Boolean value indicating whether two values are equal.
+    ///
+    /// Equality is the inverse of inequality. For any values `a` and `b`,
+    /// `a == b` implies that `a != b` is `false`.
+    ///
+    /// - Parameters:
+    ///   - lhs: A value to compare.
+    ///   - rhs: Another value to compare.
+    /// - Returns: A Boolean value indicating whether the values are equal.
+    public static func == (
```

Comment:
Do we really want to copy & paste the documentation for these? It seems DocC doesn't currently auto inherit the documentation but this is more of a tooling problem.

Reply:
This is to appease swift-format, we're strict about all public symbols having docs. I hope there'll be a feature in the future that allows omitting this and inherit from the protocol, but until then, this is fine.

Reply:
hm... not great as this more than 50% of the whole file and will get outdated eventually 😅 but yeah, not a blocker.

Reply:
Hopefully the eventual support could actually _warn_ you if you've documented a symbol documented outside your type. That way it'd tell you all the things you can remove.

Reply:
I disagree. Sometimes you want and need to overwrite the documentation. A good example is the `subscript(position: Index) -> Element` form `Collection`. You usually want to state the time complexity of the subscript as it defaults to `O(n)` if it is better (or worse).

Reply:
I'd probably instead put that info into the docs of the type, not the method coming from e.g. a protocol as I'd consider performance characteristics part of the protocol's API, and any divergence should be known at value creation time, because generics/existentials allow you to use different types the same way. So when I'm calling this method, I might not know which type I have - the protocol definition is all I have when deciding how to use the value.

But this is pretty nuanced and sure, an option to warn on documented overrides would surely be optional, not required 🙂

Reply:
`@inlinable` is missing.

Reply:
https://github.com/apple/swift-openapi-runtime/pull/58/commits/9f1e5cdd83c2e434c8c33b8d6dbf9895f19a7f70